### ISSUE TITLE
Labels ignored in folders containing /images/

### ISSUE THREAD

**Issue by agentmorris** [ultralytics/ultralytics#21224]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Train

### Bug

When generating the label filename for an image file, the [img2label_paths](https://github.com/ultralytics/ultralytics/blob/4e962d7bcbdbe3b1179d97836fd721fc376738b6/ultralytics/data/utils.py#L44) function replaces /images/ with /labels/, and replaces the extension with .txt.  This means that if I have my images organized as:

a/b/c/images/image001.jpg
a/b/c/images/image001.txt

Labels will be silently ignored.  This is obvious when *all* of your images have this property, so no labels will get picked up at all, but I started training with a folder that had the structure:

```
# About half of my images
some/folder/image001.jpg
some/folder/image001.txt
# The other half of my images
another/folder/images/image002.jpg
another/folder/images/image002.txt

```
...and it worked just well enough to convince me that everything was fine for a while, but it was silently missing the labels in the second folder, because it was looking for the labels in:

some/other/folder/labels/image002.txt

This was clear because - fortunately - a few of the impacted images happened to appear in the batch...jpg images, and it was also evident in the trained model, but this is very easy to miss in most cases.

At the cost of a potentially large number of file existence checks, it seems like preferred behavior would be to turn x/y/z/images/image001.txt into x/y/z/labels/images.txt _only if the former doesn't exist_.

I'm happy to submit a PR for this, but it would be helpful to have a gauge on whether the team considers this a bug.  To be fair, the img2label_paths function is very clear on what it does:

`"""Convert image paths to label paths by replacing 'images' with 'labels' and extension with '.txt'."""`

...so it's possible this is by design, but the "very silent failure" aspect of this seems potentially problematic, and "images" is a fairly common path name string.

### Environment

Ultralytics YOLOv8.0.215 🚀 Python-3.10.12 torch-2.1.1+cu121 CUDA:0 (NVIDIA GeForce RTX 3090, 24260MiB)
Setup complete ✅ (24 CPUs, 62.5 GB RAM, 21.9/48.9 GB disk)

OS                  Linux-6.8.0-60-generic-x86_64-with-glibc2.35
Environment         Linux
Python              3.10.12
Install             pip
RAM                 62.49 GB
CPU                 Intel Core(TM) i9-10920X 3.50GHz
CUDA                12.1

matplotlib          ✅ 3.8.2>=3.3.0
numpy               ✅ 1.26.2>=1.22.2
opencv-python       ✅ 4.8.1.78>=4.6.0
pillow              ✅ 10.1.0>=7.1.2
pyyaml              ✅ 6.0.1>=5.3.1
requests            ✅ 2.31.0>=2.23.0
scipy               ✅ 1.11.4>=1.4.1
torch               ✅ 2.1.1>=1.8.0
torchvision         ✅ 0.16.1>=0.9.0
tqdm                ✅ 4.66.1>=4.64.0
pandas              ✅ 2.1.3>=1.1.4
seaborn             ✅ 0.13.0>=0.11.0
psutil              ✅ 5.9.6
py-cpuinfo          ✅ 9.0.0
thop                ✅ 0.1.1-2209072238>=0.1.1

### Minimal Reproducible Example

See file structure above.

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @agentmorris, thank you for your detailed report and for using Ultralytics 🚀! This is an automated response to help you get started—an Ultralytics engineer will review your issue and assist you soon.

We appreciate your clear description and example file structure. If you have a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) (MRE) such as a small, shareable sample of your directory layout or code snippet that reproduces the label path issue, please provide it. This will help us debug and address your report more efficiently.

For reference, many common usage patterns and troubleshooting steps are covered in our [Docs](https://docs.ultralytics.com/). If your question relates to custom training, please include any relevant dataset image examples and training logs, and review our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community where it suits you best:
- For real-time chat, visit [Discord](https://discord.com/invite/ultralytics) 🎧
- For in-depth discussions, try [Discourse](https://community.ultralytics.com/)
- To share knowledge, check out our [Subreddit](https://www.reddit.com/r/Ultralytics/)

## Upgrade

Please ensure you are using the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/):

```bash
pip install -U ultralytics
```

## Environments

YOLO can run in any of these verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda-zone)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

Thank you again for your contribution to improving Ultralytics! 🏆

**Comment by Y-T-G** [comment]
Images without any labels are counted as "background" and the count is shown in the training logs before the training starts. So you should be able to tell from an unusual count of background images that there are labels that are not being picked up.
